Add io.github.alamahant.TarotCaster

alamahant opened the pull request:
<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

- [X] Please describe the application briefly. TarotCaster is an AI-powered tarot card reading application that combines traditional tarot wisdom with modern AI technology. It features interactive tarot card readings, built-in traditional card meanings, and AI-powered interpretations.
- [X] The domain used for the application ID is controlled by the application developer(s) and the application id guidelines are followed.
- [X] I have read and followed all the Submission requirements and the Submission guide.
- [X] I have built and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**

<!-- 💡 Mention any additional maintainers needed below 💡 -->

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

hfiguiere commented on io.github.alamahant.TarotCaster.yaml:
```diff
@@ -0,0 +1,35 @@
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
it's either `x11` alone or `fallback-x11` + `wayland`

hfiguiere commented on io.github.alamahant.TarotCaster.yaml:
```diff
@@ -0,0 +1,35 @@
+  - --filesystem=host:ro
+modules:
+  - name: tarotcaster
+    buildsystem: cmake
```
```suggestion
    buildsystem: cmake-ninja
```

hfiguiere commented on io.github.alamahant.TarotCaster.yaml:
```diff
@@ -0,0 +1,35 @@
+        tag: v1.0.0  # Use a specific tag or commit
+
+
+    build-commands:
```
this should be `post-install`

hfiguiere replied:
also all of this should be done by the cmake build system.

hfiguiere commented on io.github.alamahant.TarotCaster.yaml:
```diff
@@ -0,0 +1,35 @@
+      - install -Dm644 io.github.alamahant.TarotCaster.metainfo.xml -t /app/share/metainfo/
+      
+      # Install icon
+      - install -Dm644 io.github.alamahant.TarotCaster.png -t /app/share/icons/hicolor/512x512/apps/
```
it doesn't exist.

hfiguiere commented on io.github.alamahant.TarotCaster.desktop:
why was this added?

hfiguiere commented on io.github.alamahant.TarotCaster.metainfo.xml:
why was this added?

hfiguiere commented on io.github.alamahant.TarotCaster.png:
why was this added? It should be upstream like the rest.

hfiguiere commented on io.github.alamahant.TarotCaster.yaml:
```diff
@@ -0,0 +1,35 @@
+  - --socket=x11
+  - --socket=wayland
+  - --device=dri
+  - --filesystem=host:ro
```
this is a very broad filesystem permission, why is it needed?

alamahant commented:
Thank you for your thorough review! I've made the following changes to
address all your feedback:
1. Socket permissions: Changed from separate x11 and wayland to
fallback-x11 and wayland
2. Build system: Updated to use cmake-ninja instead of cmake
3. Desktop integration files: I've moved all desktop integration files
(desktop entry, metainfo, and icon) to my upstream repository and
updated the CMakeLists.txt to properly install them. These files are
no longer needed in the Flathub repository.
4. Installation commands: I've updated my upstream CMakeLists.txt to
handle all installation tasks properly, so the manual build-commands
are no longer needed in the manifest.
5. Filesystem permission: I've removed the broad --filesystem=host:ro
permission as it's not necessary for the application to function.

I've created a new release (v1.0.1) with these changes and updated the
manifest to use this new release. The manifest is now much simpler and
relies on the upstream build system to handle all installation tasks.

Thank you for helping me improve my submission!


On Wed, Apr 9, 2025 at 2:01 AM Hubert Figuière ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In io.github.alamahant.TarotCaster.yaml
> <https://github.com/flathub/flathub/pull/6357#discussion_r2034143497>:
>
> > @@ -0,0 +1,35 @@
> +app-id: io.github.alamahant.TarotCaster
> +runtime: org.kde.Platform
> +runtime-version: "6.8"
> +sdk: org.kde.Sdk
> +command: TarotCaster
> +finish-args:
> +  - --share=network
> +  - --share=ipc
> +  - --socket=x11
> +  - --socket=wayland
> +  - --device=dri
> +  - --filesystem=host:ro
>
> this is a very broad filesystem permission, why is it needed?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flathub/flathub/pull/6357#pullrequestreview-2751578850>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AELIXCSOGYKCVY4N7ONSGHD2YRILJAVCNFSM6AAAAAB2XOLRZ2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDONJRGU3TQOBVGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

hfiguiere commented:
bot, build io.github.alamahant.TarotCaster

alamahant commented:
Just tested the build, everything is working smoothly! Thanks !

alamahant commented:
I just tested the build.And iit is working as expected.Thanks a lot!

On Wed, Apr 9, 2025 at 4:52 AM Flathub Bot ***@***.***> wrote:

> Build 192453 successful
> <https://buildbot.flathub.org/#/builders/6/builds/192453>
> To test this build, install it from the testing repository:
>
> flatpak install --user https://dl.flathub.org/build-repo/175741/io.github.alamahant.TarotCaster.flatpakref
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flathub/flathub/pull/6357#issuecomment-2788076081>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AELIXCX4VSHPC7EOO3AEJHT2YR4M7AVCNFSM6AAAAAB2XOLRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOBYGA3TMMBYGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> *flathubbot* left a comment (flathub/flathub#6357)
> <https://github.com/flathub/flathub/pull/6357#issuecomment-2788076081>
>
> Build 192453 successful
> <https://buildbot.flathub.org/#/builders/6/builds/192453>
> To test this build, install it from the testing repository:
>
> flatpak install --user https://dl.flathub.org/build-repo/175741/io.github.alamahant.TarotCaster.flatpakref
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flathub/flathub/pull/6357#issuecomment-2788076081>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AELIXCX4VSHPC7EOO3AEJHT2YR4M7AVCNFSM6AAAAAB2XOLRZ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOBYGA3TMMBYGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

bbhtt commented:
/merge
